### PR TITLE
Bugfix: make registry_key resource case-insensitive

### PR DESCRIPTION
**Pull request by alexpop**
Add integration test for `NtlmMinClientSec` 

Once https://github.com/chef/inspec/issues/254 is fixed, we should add a test for `NTLMMinServerSec`

### CODE REVIEW

**Comment by chris-rock** [comment]
Thanks for adding this @alexpop

**Comment by chris-rock** [comment]
@arlimus added fix to this PR

File: lib/resources/registry_key.rb [hunk 1]
```diff
@@ -87,11 +104,13 @@ def registry_value(path)
     # return nil if cmd.exit_status != 0, try to parse json
     begin
       @registy_cache = JSON.parse(cmd.stdout)
+      # convert keys to lower case
+      return @registy_cache = Hash[@registy_cache.map do |key, value|
+        [key.downcase, value]
+      end]
```
Comment (arlimus) [inline lib/resources/registry_key.rb:110]: could we:

```
begin
  keys = JSON.parse(cmd.stdout)
  @registry_cache = ... blah ...
rescue JSON::ParseError => _
  @registry_cache = nil
end

@registry_cache
```

This also avoids a duplicate return statement.

Comment (chris-rock) [inline lib/resources/registry_key.rb:110]: fixed

**Comment by arlimus** [comment]
Thank you @alexpop , this is an awesome addition! Thanks @chris-rock for extending our tests.

**Pull request merged**